File: C#_Base_PT4/XString28.cs
// File: "XString28"
using PT4;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace PT4Tasks
{
    public class MyTask : PT
    {
        public static void Solve()
        {
            Task("XString28");
            char c = GetChar();
            string s = GetString();
            s = s.Replace(new string(c,1), new string (c,2) );
            Put(s);
        }
    }
}

File: C++_Base_PT4/XString28.cpp
#include "pt4.h"
using namespace std;

void Solve()
{
    Task("XString28");
    
    char c;
	
	pt >> c;
	
	string str;
	
	pt >> str;
	
	int n = 0;
	
	int i = 0;
	
	while(1)
	{
		n = str.find(c,i);
		
		if(n < 0) break;
		
		str=str.replace(n,1,2,c);
		
		i=n+2;
		
	}
	
	pt << str;
	
}
